AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -211,18 +202,32 @@ - (BOOL)validateLogName:(NSString *)name {
   return YES;
 }
 
-- (BOOL)validateLogData:(MSCSData *)data {
-  NSDictionary<NSString *, NSString *> *properties = data.properties;
+- (NSDictionary<NSString *, id> *)validateProperties:(NSDictionary<NSString *, id> *)properties {
+  NSMutableDictionary<NSString *, id>
+      *validProperties = [NSMutableDictionary new];
   for (NSString *key in properties) {
-    if (![key isKindOfClass:[NSString class]] ||
-        ![properties[key] isKindOfClass:[NSString class]]) {
-      MSLogError([MSAppCenter logTag],
-                 @"%@ Properties key and value must be of type NSString.",
-                 kMSBaseErrorMsg);
-      return NO;
+    BOOL keyIsValid = NO;
+    if ([key isKindOfClass:[NSString class]] && key.length > 0) {
+      keyIsValid = YES;
```

Comment:
If a key isn't valid, you can drop the key and continue. I think you can simply `continue` here.

Reply:
I think it might be a miscommunication.
What I meant was that you don't even need `keyIsValid` variable.
```
if ([key isKindOfClass:[NSString class]] && key.length > 0) {
   id value = properties[key];
   if (value) {
       // Not checking for empty string, as values can be empty strings.
      if ([value isKindOfClass:[NSString class]]) {
          [validProperties setValue:value forKey:key];
      } else if ([value isKindOfClass:[NSDictionary class]]) {
        NSDictionary *nestedValidProperties = [self validateProperties:value];
        [validProperties setValue:nestedValidProperties forKey:key];
      }
   }
}